aten/src/ATen/native/mps/operations/SummaryOps.mm

```diff
@@ -0,0 +1,146 @@
+#include <ATen/ATen.h>
+#include <ATen/Tensor.h>
+#include <ATen/Utils.h>
+#include <ATen/mps/MPSStream.h>
+#include <ATen/native/mps/OperationUtils.h>
+#include <torch/library.h>
+#include <c10/util/Optional.h>
+
+namespace at {
+namespace native {
+
+Tensor& bincount_mps_impl(const Tensor& self,
+                          const Tensor& weights,
+                          Tensor& output) {
+  using namespace mps;
+
+  struct CachedGraph : public MPSCachedGraph
+  {
+    CachedGraph(MPSGraph *graph) : MPSCachedGraph(graph) {}
+    MPSGraphTensor* inputTensor_ = nil;
+    MPSGraphTensor* weightsTensor_ = nil;
+    MPSGraphTensor* scatterDataTensor_ = nil;
+    MPSGraphTensor* outputTensor_ = nil;
+  };
+
+  MPSStream* stream = getCurrentMPSStream();
+  MPSGraphCache* cache_ = MPSGraphCache::getInstance();
+  bool has_weights = weights.defined();
+
+  @autoreleasepool {
+    string key =  "bincount_mps_impl:" +
+            (has_weights ? getTensorsStringKey({self, weights}) : getTensorsStringKey(self));
```

Comment:
- You don't need the has_weights check in here, and just `getTensorsStringKey({self, weights})` would be enough. If weights.defined() is false, the getTensorsStringKey() will put the constant string "Undefined" in its place in the string key.
- Also you don't need the ':' after "bincount_mps_impl". The getTensorsStringKey() will start with ':'.

Reply:
Thanks for the suggestion - I'll update this